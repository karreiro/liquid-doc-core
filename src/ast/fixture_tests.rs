// In your test file
include!(concat!(env!("OUT_DIR"), "/fixtures_macro.rs"));

macro_rules! generate_fixture_test {
    ($fixture_name:expr) => {
        paste::paste! {
            #[test]
            fn [<test_fixture_ $fixture_name>]() {
                let content = include_str!(concat!("../../web/fixtures/", $fixture_name, ".liquid"));
                // Your test logic here
                crate::assert_json_output!(content);
            }
        }
    };
}

// This will generate all the tests
for_each_fixture_file!(generate_fixture_test);
