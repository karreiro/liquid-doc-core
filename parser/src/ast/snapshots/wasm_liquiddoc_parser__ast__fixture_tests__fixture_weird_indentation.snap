---
source: src/ast/fixture_tests.rs
expression: "this block was AI generated\n        @prompt\n          First prompt\n        @param {String} paramName - param description"
---
[
  {
    "type": "LiquidDocDescriptionNode",
    "name": "description",
    "position": {
      "start": 10,
      "end": 38
    },
    "source": "this block was AI generated\n",
    "content": {
      "type": "TextNode",
      "value": "this block was AI generated\n",
      "position": {
        "start": 10,
        "end": 38
      },
      "source": "this block was AI generated\n"
    },
    "isImplicit": true,
    "isInline": true
  },
  {
    "type": "TextNode",
    "value": "@prompt\n          First prompt\n",
    "position": {
      "start": 46,
      "end": 77
    },
    "source": "@prompt\n          First prompt\n"
  },
  {
    "type": "LiquidDocParamNode",
    "name": "param",
    "position": {
      "start": 85,
      "end": 130
    },
    "source": "@param {String} paramName - param description",
    "paramName": {
      "type": "TextNode",
      "value": "paramName",
      "position": {
        "start": 101,
        "end": 110
      },
      "source": "paramName"
    },
    "paramDescription": {
      "type": "TextNode",
      "value": "param description",
      "position": {
        "start": 113,
        "end": 130
      },
      "source": "param description"
    },
    "paramType": {
      "type": "TextNode",
      "value": "String",
      "position": {
        "start": 93,
        "end": 99
      },
      "source": "{String}"
    },
    "required": true
  }
]
